### DISCUSSION TITLE
MPP is not working on RK3566 platform

### DISCUSSION THREAD

**Discussion by gilmotta3** [rockchip-linux/mpp#691] [General]
Hello friends,

I cloned the latest MPP repo and built it according the following instructions:

git clone https://github.com/rockchip-linux/mpp
pushd mpp
cmake -DRKPLATFORM=ON -DHAVE_DRM=ON && make -j$(nproc)
sudo make install
sudo cp ./mpp/librockchip_mpp.so* /lib/aarch64-linux-gnu
sudo cp ./mpp/legacy/librockchip_vpu.so* /lib/aarch64-linux-gnu
popd

then I built FFmpeg accoding the following instructions:

git clone https://github.com/hbiyik/FFmpeg.git
pushd FFmpeg/
./configure --enable-rkmpp --enable-version3 --enable-libdrm --enable-libx264 --enable-gpl
make -j$(nproc)
sudo make install
popd

However FFmpeg is not able to initialize MPP and it is throwing an error!
`
linearity@kilasRK4G:~/development/media$ ffmpeg -v verbose -re -i big_buck_bunny_1080p_h264.mov -c:v hevc -rc_mode 0 -level 30 -f rtsp rtsp://0.0.0.0:8554/stream
ffmpeg version 00a3a09a7e Copyright (c) 2000-2023 the FFmpeg developers
  built with gcc 10 (Debian 10.2.1-6)
  configuration: --enable-rkmpp --enable-version3 --enable-libdrm --enable-libx264 --enable-gpl
  libavutil      58.  2.100 / 58.  2.100
  libavcodec     60.  3.100 / 60.  3.100
  libavformat    60.  3.100 / 60.  3.100
  libavdevice    60.  1.100 / 60.  1.100
  libavfilter     9.  3.100 /  9.  3.100
  libswscale      7.  1.100 /  7.  1.100
  libswresample   4. 10.100 /  4. 10.100
  libpostproc    57.  1.100 / 57.  1.100
[h264 @ 0x5584601b50] Reinit context to 1920x1088, pix_fmt: yuv420p
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'big_buck_bunny_1080p_h264.mov':
  Metadata:
    major_brand     : qt
    minor_version   : 537199360
    compatible_brands: qt
    creation_time   : 2008-05-27T18:40:35.000000Z
    com.apple.quicktime.player.movie.audio.gain: 1.000000
    com.apple.quicktime.player.movie.audio.treble: 0.000000
    com.apple.quicktime.player.movie.audio.bass: 0.000000
    com.apple.quicktime.player.movie.audio.balance: 0.000000
    com.apple.quicktime.player.movie.audio.pitchshift: 0.000000
    com.apple.quicktime.player.movie.audio.mute:
    com.apple.quicktime.player.movie.visual.brightness: 0.000000
    com.apple.quicktime.player.movie.visual.color: 1.000000
    com.apple.quicktime.player.movie.visual.tint: 0.000000
    com.apple.quicktime.player.movie.visual.contrast: 1.000000
    com.apple.quicktime.player.version: 7.4.1 (14)
    com.apple.quicktime.version: 7.4.1 (14) 0x7418000 (Mac OS X, 10.5.2, 9C31)
    timecode        : 00:00:00:00
  Duration: 00:09:56.46, start: 0.000000, bitrate: 9725 kb/s
  Stream #0:0[0x1](eng): Video: h264 (Main), 1 reference frame (avc1 / 0x31637661), yuv420p(tv, bt709, progressive, topleft), 1920x1080 (1920x1088), 9282 kb/s, 24 fps, 24 tbr, 2400 tbn (default)
    Metadata:
      creation_time   : 2008-05-27T18:40:35.000000Z
      handler_name    : Apple Video Media Handler
      vendor_id       : appl
      encoder         : H.264
    Side data:
      displaymatrix: rotation of -0.00 degrees
  Stream #0:1[0x2](eng): Data: none (tmcd / 0x64636D74) (default)
    Metadata:
      creation_time   : 2008-05-27T18:40:35.000000Z
      handler_name    : Time Code Media Handler
      timecode        : 00:00:00:00
  Stream #0:2[0x3](eng): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, 5.1, fltp, 437 kb/s (default)
    Metadata:
      creation_time   : 2008-05-27T18:40:35.000000Z
      handler_name    : Apple Sound Media Handler
      vendor_id       : [0][0][0][0]
[vost#0:0 @ 0x55848fa6f0] Matched encoder 'hevc_rkmpp_encoder' for codec 'hevc'.
[h264_rkmpp_decoder @ 0x5584941710] Picture format is nv12.
[h264_rkmpp_decoder @ 0x5584941710] Failed to initialize MPP context (code = -1).
[h264_rkmpp_decoder @ 0x5584941710] Failed to initialize RKMPP Codec.
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (h264_rkmpp_decoder) -> hevc (hevc_rkmpp_encoder))
  Stream #0:2 -> #0:1 (aac (native) -> aac (native))
Error while opening decoder for input stream #0:0 : Unknown error occurred
[AVIOContext @ 0x5584609170] Statistics: 509922 bytes read, 2 seeks
linearity@kilasRK4G:~/development/media$
`

and from the log I only see one line which I have no clue!

Oct  8 16:30:10 kilasRK4G mpp[26762]: mpp_info: mpp version: 4a71a39d author: timkingh.huang 2024-09-25 fix[smt_rc]: Fix first frame QP error

Can anyone please assist me with this? My goal is to run hardware encoding on the RK3566. I would l like to use ffmpeg and gstreamer on the RK3566 to encode the video input.

Thank you!

**Comment by HermanChen** [comment]
https://github.com/nyanmisaka/ffmpeg-rockchip
try this repo

**Reply by gilmotta3** [reply]
Hello Herman! Thank you , I will try and let you know.
regards,
